app/helpers.php

```diff
@@ -93,17 +93,28 @@ function getNorthstarId($request)
 
 /**
  * Determines if the user is an admin or staff.
+ *
+ * @return bool
  */
-function is_staff_user()
+function is_staff_user() : bool
 {
-    $isStaffUser = token()->exists() && in_array(token()->role, ['admin', 'staff']);
-    $isStaffClient = token()->exists() && in_array('admin', token()->scopes());
-
-    if ($isStaffUser || $isStaffClient) {
+    // If this is a machine client, then it's de-facto an admin:
+    if (token()->exists() && ! token()->id()) {
```

Comment:
Is there any way this could be abused?

Reply:
I don't believe so. The only way you should be able to get a token without an ID attached is via the client credentials grant, which is only used by internal "machine" applications.

Reply:
Someone couldn't delete the ID part from their token then? It all comes bundled together?

Reply:
Ah nope! If the user were to modify [their token](https://github.com/DoSomething/northstar/blob/master/documentation/authentication.md#access-tokens), the signature would become invalid & refused by the server.

Reply:
🔒